react/Footer/FooterLink/FooterLink.js

```diff
@@ -36,9 +38,16 @@ FooterLink.propTypes = {
     PropTypes.arrayOf(PropTypes.node),
     PropTypes.node
   ]),
-  linkRenderer: PropTypes.func.isRequired
+  linkRenderer: PropTypes.func.isRequired,
+  authedHref: PropTypes.string,
```

Comment:
This api seems a bit unusual to me. I feel there may be a more generic solution here. Maybe `authRequired` as a boolean and based on this is will either honour your `href` or set it as a returnUrl for the sign-in path. Thoughts?